File: frontend/src/types/transaction.ts
export interface Transaction {
  transaction_id: string;
  client_id: string;
  isin: string;
  action: "buy" | "sell";
  quantity: number;
  price: number;
  timestamp: string;
  upload_batch_id: string;
}

export interface UploadError {
  row: number;
  field: string;
  message: string;
}

export interface UploadResult {
  batch_id: string;
  total_rows: number;
  valid_rows: number;
  error_rows: number;
  errors: UploadError[];
  affected_clients: string[];
}

File: backend/app/schemas/transaction.py
"""Transaction request/response schemas."""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class TransactionOut(BaseModel):
    transaction_id: str
    client_id: str
    isin: str
    action: str
    quantity: Decimal
    price: Decimal
    timestamp: datetime
    upload_batch_id: str

    model_config = {"from_attributes": True}


class UploadError(BaseModel):
    row: int
    field: str
    message: str


class UploadResult(BaseModel):
    batch_id: str
    total_rows: int
    valid_rows: int
    error_rows: int
    errors: list[UploadError]
    affected_clients: list[str]
